File: frontend/web/features/management/types/department.ts
export type DepartmentValues = Record<string, string> & {
  departmentCode: string;
  name: string;
  headTeacherId: string;
  head: string;
  status: string;
  createAt: string;
};

export type DepartmentItem = { id: string; values: DepartmentValues };

File: backend/src/InstituteManagement.Domain/Entities/Management/Departments/Department.cs
namespace InstituteManagement.Domain.Entities;

public sealed class Department : Entity
{
    public required string DepartmentCode { get; set; } = string.Empty;
    public required string Name { get; set; }
    public string Head { get; set; } = string.Empty;
    public Guid? HeadTeacherId { get; set; }
    public Teacher? HeadTeacher { get; set; }
    public bool IsActive { get; set; } = true;
}
